build.gradle

```diff
@@ -1,17 +1,38 @@
+plugins {
+    id "com.diffplug.spotless" version "5.12.4"
+}
+
 apply from: file("${rootDir}/build-resources.gradle")
 allprojects {
+    apply plugin: 'com.diffplug.spotless'
+
     group = 'com.amazon'
     version = '0.8.0-beta'
     repositories {
         mavenCentral()
         maven { url "https://jitpack.io" }
     }
+
+    spotless {
+        format 'misc', {
+            target '*.md', '.gitignore', '*.yml', '*.yaml'
+            // TODO: enrich format rules
+            trimTrailingWhitespace()
```

Comment:
We might want to _not_ trim whitespace on markdown files. From https://daringfireball.net/projects/markdown/syntax#block:

>"When you do want to insert a \<br /> break tag using Markdown, you end a line with two or more spaces, then type return."

Would just need to confirm if the formatter would strip double-spaces which were intended to be used as line breaks. Or we can just encourage people to not use double-spaces in our files. 😉 

I only see one double-space being removed in this PR, and I don't think it was intended to be a line break. Can probably merge as is and test locally.

Reply:
> We might want to not trim whitespace on markdown files. From https://daringfireball.net/projects/markdown/syntax#block:

Sure. Will reserve markdown in a separate block.

> Would just need to confirm if the formatter would strip double-spaces which were intended to be used as line breaks. Or we can just encourage people to not use double-spaces in our files.

No. Tested it does not.